src/detect-dataset.c

```diff
@@ -406,10 +406,6 @@ int DetectDatasetSetup (DetectEngineCtx *de_ctx, Signature *s, const char *rawst
         SCLogError("failed to set up dataset '%s'.", name);
         return -1;
     }
-    if (set->hash && SC_ATOMIC_GET(set->hash->memcap_reached)) {
```

Comment:
What about detect-data rep.c ?

Reply:
that should be covered as well since this also uses the `DatasetGet` function and would now benefit from the same handling within `src/dataset.c` so even a good additional argument to move it like we plan to :)

Reply:
Looks good indeed.

Should that be reflected in the commit message ?

Reply:
> I think it's already quite generic to datasets

From what I understand, it is different :
- this commit adds the check for datarep, it did not exist before (it should be testable by suricata-verify)
- this commit moves the check for dataset to avoid dangling memory leaking before final cleanup

If this is correct, I think the commit message should reflect the behavior change for the datarep keyword...